packages/dialog/src/dialog-wrapper.ts

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+    html,
+    LitElement,
+    TemplateResult,
+    property,
+    CSSResultArray,
+} from 'lit-element';
+import { ifDefined } from 'lit-html/directives/if-defined';
+
+import '@spectrum-web-components/underlay';
+
+import styles from './dialog-wrapper.css.js';
+
+/**
+ * @element sp-dialog-wrapper
+ *
+ * @fires secondary - Announces that the "secondary" button has been clicked.
+ * @fires cancel - Announces that the "cancel" button has been clicked.
+ * @fires confirm - Announces that the "confirm" button has been clicked.
+ * @fires close - Announces that the dialog has been closed.
+ */
+export class DialogWrapper extends LitElement {
+    public static get styles(): CSSResultArray {
+        return [styles];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public error = false;
+
+    @property({ attribute: 'cancel-label' })
+    public cancelLabel = '';
+
+    @property({ attribute: 'confirm-label' })
+    public confirmLabel = '';
+
+    @property({ type: Boolean, reflect: true })
+    public dismissible = false;
+
+    @property()
+    public footer = '';
+
+    @property()
+    public hero = '';
+
+    @property({ type: Boolean, reflect: true, attribute: 'no-divider' })
+    public noDivider = false;
+
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    @property({ type: String, reflect: true })
+    public mode: 'fullscreen' | 'fullscreenTakeover' | '' = '';
+
+    @property({ type: String, reflect: true })
+    public size: 'small' | 'medium' | 'large' | 'alert' | '' = '';
```

Comment:
Setting the default size to 'blank' is a but confusing for me. Is that equivalent to 'medium'?

Reply:
DId I write that? Hrm 🤔 It's the equivalent of it not being there...let me look deeper at what that does, and how it interacts with the styles. In some cases I've used a more complex getter/setter pair to ensure that the attribute was removed in this case, would that be more or less confusing?

Reply:
As we're not using this internally (beyond presence testing), would it be less confusing for this to be types as follows?
```
public size?: 'small' | 'medium' | 'large' | 'alert';
```

Reply:
The above looks more clear to me, thanks! 

So a size of null is equivalent to the dialog not being active, that makes sense to me 👍

Reply:
`<sp-dialog open size="small|medium|large|alert">` enforces specific widths onto the dialog.

`<sp-dialog open>` will be open and take up as much space as it needs.

`<sp-dialog>` will not be active.

Reply:
Catch me up on how this feels now that that most recent code is up and passing CI.

Reply:
This API looks solid to me 👍